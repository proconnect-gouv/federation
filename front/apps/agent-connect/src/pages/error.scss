div.content-error {
  max-width: 600px;
  margin-left: auto;
  margin-right: auto;
}