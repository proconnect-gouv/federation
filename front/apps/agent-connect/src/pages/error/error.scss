.ErrorComponent {
  max-width: 600px;
}
