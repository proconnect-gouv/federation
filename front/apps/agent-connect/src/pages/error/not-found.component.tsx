import React from 'react';

export const NotFoundComponent = React.memo(() => (
  <div className="container">
    <h1 className="text-center">404 - Not Found</h1>
  </div>
));

NotFoundComponent.displayName = 'NotFoundComponent';
