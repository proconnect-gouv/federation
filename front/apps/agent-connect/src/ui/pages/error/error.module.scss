.page {
  max-width: 600px;
}

.title {
  color: var(--blue-agent-connect);
}

.details {
  background: var(--background-alt-grey);
}
