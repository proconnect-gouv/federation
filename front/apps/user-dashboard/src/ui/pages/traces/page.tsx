import Introduction from './introduction';
import TracesList from './traces-list';
import UserWelcome from './user-welcome';

const TracesPageComponent = () => (
  <div className="content-wrapper-lg px16" id="page-container">
    <UserWelcome />
    <Introduction />
    <TracesList />
  </div>
);

TracesPageComponent.displayName = 'TracesPageComponent';

export default TracesPageComponent;
