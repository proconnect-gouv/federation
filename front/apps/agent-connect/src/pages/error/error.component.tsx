import './error.scss';

import React from 'react';

import classnames from 'classnames';
import { useMediaQuery } from 'react-responsive';
import { IError } from '../../types/error.type';

interface ErrorComponentProps {
  errors: IError;
}

export const ErrorComponent = React.memo(({ errors }: ErrorComponentProps) => {
  const gtTablet = useMediaQuery({ query: '(min-width: 768px)' });

  return (
    <div>
      <h1
        className={classnames('is-blue-agentconnect is-bold px16', {
          'text-center': gtTablet,
          'text-left': !gtTablet,
        })}>
        Une erreur est survenue lors de la connexion.
      </h1>
      <div className="content-error my32 px16 fr-text-lg">
        <p className="mb32 is-bold">Description de l’erreur :</p>
        <p className="mb32" id="error-message">
          {errors.message}
        </p>
        <p className="mb32 is-bold">Que faire ? </p>
        <p className="mb32">
          Fermez l’onglet de votre navigateur et reconnectez-vous en cliquant sur le bouton
          AgentConnect.
        </p>
        <p className="mb32">Contactez le support de votre service si le problème persiste.</p>
        <div className="bg-blue-france-100 p24">
          <p>
            Informations à nous transmettre dans le mail pour faciliter la prise en charge de votre
            demande :
          </p>
          <p id="error-code">
            <strong>Code d’erreur :</strong> {errors.code}
          </p>
          <p>
            <strong>ID :</strong> {errors.id}
          </p>
        </div>
      </div>
    </div>
  );
});

ErrorComponent.displayName = 'ErrorComponent';
