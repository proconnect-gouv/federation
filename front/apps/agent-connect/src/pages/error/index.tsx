import React from 'react';

import { ErrorComponent } from './error.component';
import { NotFoundComponent } from './not-found.component';

export const ErrorPage = React.memo(() => {
  const hasError = (window as any).appError !== undefined;
  if (hasError) {
    return <ErrorComponent errors={(window as any).appError} />;
  }
  return <NotFoundComponent />;
});

ErrorPage.displayName = 'ErrorPage';
