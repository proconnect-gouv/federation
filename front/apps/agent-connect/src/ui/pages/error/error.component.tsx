import classnames from 'classnames';
import React from 'react';
import { useMediaQuery } from 'react-responsive';

import { IError } from '../../../types/error.type';
import styles from './error.module.scss';

interface ErrorComponentProps {
  errors: IError;
}

export const ErrorComponent = React.memo(({ errors }: ErrorComponentProps) => {
  const gtTablet = useMediaQuery({ query: '(min-width: 768px)' });

  return (
    <div
      className={classnames(styles.page, 'fr-m-auto', {
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-4w': !gtTablet,
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-8w': gtTablet,
      })}>
      <h1
        className={classnames(styles.title, 'fr-text--bold fr-px-2w', {
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'text-center': gtTablet,
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'text-left': !gtTablet,
        })}>
        Une erreur est survenue lors de la connexion.
      </h1>
      <div className="fr-mx-auto fr-my-4w fr-px-2w fr-text--lg">
        <p className="fr-mb-4w fr-text--bold">Description de l’erreur :</p>
        <p className="fr-mb-8w" data-testid="error-message">
          {errors.message}
        </p>
        <p className="fr-mb-4w fr-text--bold">Que faire ?</p>
        <p className="fr-mb-4w">
          Fermez l’onglet de votre navigateur et reconnectez-vous en cliquant sur le bouton
          AgentConnect.
        </p>
        <p className="fr-mb-4w">Contactez le support de votre service si le problème persiste.</p>
        <div className={classnames(styles.details, 'fr-p-3w')}>
          <p>
            Informations à nous transmettre dans le mail pour faciliter la prise en charge de votre
            demande :
          </p>
          <p data-testid="error-code">
            <strong>Code d’erreur :</strong> {errors.code}
          </p>
          <p className="fr-my-0" data-testid="error-session-id">
            <strong>ID :</strong> {errors.id}
          </p>
        </div>
      </div>
    </div>
  );
});

ErrorComponent.displayName = 'ErrorComponent';
