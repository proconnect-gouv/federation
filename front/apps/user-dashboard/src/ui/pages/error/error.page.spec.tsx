// @TODO check if should be completed
import { render } from '@testing-library/react';

import { ErrorPage } from './error.page';

describe('ErrorPage', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    jest.restoreAllMocks();
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    // when
    const { container } = render(<ErrorPage />);
    // expect
    expect(container).toMatchSnapshot();
  });
});
