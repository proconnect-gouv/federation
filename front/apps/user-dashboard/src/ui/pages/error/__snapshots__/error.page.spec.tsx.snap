// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should match the snapshot 1`] = `
<div>
  <div
    class="fr-m-auto fr-px-2w text-center fr-mt-5w"
    id="page-container"
  >
    <h1
      class="text-center"
    >
      Une erreur est survenue
    </h1>
    <h2
      class="text-center"
    >
      <span>
        Erreur
      </span>
    </h2>
    <p
      class="text-center"
    >
      Ceci est une erreur
    </p>
  </div>
</div>
`;
